pkg/resources/resources.go

```diff
@@ -48,26 +48,38 @@ func GetContainerImagesFromJob(job *batchv1.Job) (kube.ContainerImages, error) {
 // cluster-scoped permissions to get Nodes in order to set the owner reference
 // when we create an instance of custom security report.)
 //
+// Pods created and controlled by third party frameworks, such as Argo workflow
+// engine, are considered as unmanaged. Otherwise we'd need to maintain and
+// extend the list of RBAC permissions over time.
 // TODO Merge this method with OwnerResolver, which accepts kube.Object and resolves client.Object.
 func GetImmediateOwnerReference(pod *corev1.Pod) kube.Object {
 	ownerRef := metav1.GetControllerOf(pod)
 
-	if ownerRef != nil {
-		if ownerRef.Kind == "Node" {
-			// Static pod ~ unmanaged pod
-			return kube.Object{
-				Kind:      kube.KindPod,
-				Namespace: pod.Namespace,
-				Name:      pod.Name,
-			}
+	if ownerRef == nil {
+		return kube.Object{
+			Kind:      kube.KindPod,
+			Namespace: pod.Namespace,
+			Name:      pod.Name,
```

Comment:
This is super picky, but I don't love that the unmanaged pod version appears 3 times! Could be something like

```
if ownerRef != nil {
  switch ownerRef.Kind {
    case "Pod", "ReplicaSet"... :
      return kube.Object {
        Kind:      kube.Kind(ownerRef.Kind),
        ...
      }
  }
}

// Pod owned by anything else is treated the same as an unmanaged pod
return kube.Object {
        Kind: kube.KindPod,
        ...
}
```

Reply:
You're right. I've applied your suggestion. Reads much better now.